### PR TITLE
Prevent merge/delete operations when wals exist

### LINKED ISSUE #3845: Can not merge tablets with write ahead logs
Tablets with write ahead logs should not be merged, see #3844 for details on why.  If a tablet has write ahead logs, then merge should wait for those to be processed before proceeding.  Need to see if the existing merge code defends against this, if this may be a problem for more than the elasticity branch.

### PR DESCRIPTION
**Pull request by cshannon**
This commit updates the merge and delete rows FATE ops to verify that if tablets being processed do not have any wals

This closes #3845

This PR makes the following changes:

1. Modifies ReserveTablets FATE op to check that the tablets have no wals when reserving
2. Modifies `MergeTablets` and `DeleteRows` FATE ops to add verification that the tablets have no wals
3. Modifies `TabletManagementIterator` to return a tablet back for processing again by TGW if there's no location, an op id set and wals exist
4. Modifies `TabletGroupWatcher` to set the tablet goal to be hosted to bring a tablet online if wals exist and a fate id is also set

I have marked this as a draft for now because I believe these are all the changes needed but I wanted to explore adding a test or 2 if possible (maybe `TabletManagementIteratorIT`, etc)

### CODE REVIEW

File: server/manager/src/main/java/org/apache/accumulo/manager/tableOps/merge/ReserveTablets.java
Comment (keith-turner) [inline server/manager/src/main/java/org/apache/accumulo/manager/tableOps/merge/ReserveTablets.java:82]: Need WALS to the coumns to fetch above or this will throw an exception

File: server/manager/src/main/java/org/apache/accumulo/manager/tableOps/merge/MergeTablets.java
Comment (keith-turner) [inline server/manager/src/main/java/org/apache/accumulo/manager/tableOps/merge/MergeTablets.java:237]: For the method this call this if they are fetching columns, then need to add the WALS.   Like DeleteRows calls this and is fetching columns.

File: server/manager/src/main/java/org/apache/accumulo/manager/TabletGroupWatcher.java
Comment (keith-turner) [inline server/manager/src/main/java/org/apache/accumulo/manager/TabletGroupWatcher.java:403]: May want to limit this to hosting on merge operations with wals, so if opid is merge and it has wals then host.  Eventually split will need it too (#3844), but may be good to make the FATE aware of logs.  Delete operations probably will never need tablets hosted for recovery.

File: server/base/src/main/java/org/apache/accumulo/server/manager/state/TabletManagementIterator.java
Comment (keith-turner) [inline server/base/src/main/java/org/apache/accumulo/server/manager/state/TabletManagementIterator.java:284]: May want to limit this to merging operation for now.  Split will need this once its fully implement (#3844).  Delete table will never need it.

```suggestion
        if (tm.getLogs().isEmpty() && tm.getOperationId() != null && tm.getOperationId().getType() == TabletOperationType.MERGING) {
```

**Comment by cshannon** [comment]
@keith-turner - I made the requested changes and I think everything is correct so I marked this as ready. I started working on a test update in `TabletManagementIteratorIT` but didn't have time to finish yet as it's taking some time because I am trying to figure out how `TabletManagementIterator` works as all of that code is new to me and pretty complex. So you can either merge this now (if you think it's ok) and I can create a follow on issue to add some ITs or you can wait to merge this until I have more time later this week to try and work on some tests.

File: server/manager/src/main/java/org/apache/accumulo/manager/TabletGroupWatcher.java
Comment (keith-turner) [inline server/manager/src/main/java/org/apache/accumulo/manager/TabletGroupWatcher.java:407]: When a tablet has an operationid, generally want it unassigned.  So need to work in an exception of that in the case of walogs and merge

```suggestion
        if (!tm.getLogs().isEmpty()
            && tm.getOperationId().getType() == TabletOperationType.MERGING) {
          goal = TabletGoalState.HOSTED;
        } else {
          goal = TabletGoalState.UNASSIGNED;
        }
```

File: server/base/src/main/java/org/apache/accumulo/server/manager/state/TabletManagementIterator.java
Comment (keith-turner) [inline server/base/src/main/java/org/apache/accumulo/server/manager/state/TabletManagementIterator.java:285]:
```suggestion
        if (!tm.getLogs().isEmpty() && tm.getOperationId() != null
```

**Comment by keith-turner** [comment]
I opened #3904 to refactor the TGW and TabletMgmtIterator code to avoid having logic like what this PR is adding in two places.  After this is merged I can merge these changes into #3904 and resolve conflicts.

File: server/base/src/main/java/org/apache/accumulo/server/manager/state/TabletManagementIterator.java
Comment (cshannon) [reply server/base/src/main/java/org/apache/accumulo/server/manager/state/TabletManagementIterator.java:285]: @keith-turner - Oops, nice catch this is probably why the test i started to write didn't work as this logic was backwards. I can revisit the tests after you merge in your refactoring for #3904.

**Review by keith-turner — APPROVED** [review]

**Pull request merged**
